mindsdb_sdk/agents.py

```diff
@@ -142,6 +156,54 @@ def completion(self, name: str, messages: List[dict]) -> AgentCompletion:
         data = self.api.agent_completion(self.project, name, messages)
         return AgentCompletion(data['message']['content'])
 
+    def add_file(self, name: str, file_path: str, description: str, knowledge_base: str = None):
+        """
+        Add a file to the agent for retrieval.
+
+        :param name: Name of the agent
+        :param file_path: Path to the file to be added, or name of existing file.
+        :param description: Description of the file. Used by agent to know when to do retrieval
+        :param knowledge_base: Name of an existing knowledge base to be used. Will create a default knowledge base if not given.
+        """
+        filename = file_path.split('/')[-1]
+        filename_no_extension = filename.split('.')[0]
+        try:
+            _ = self.api.get_file_metadata(filename_no_extension)
+        except HTTPError as e:
+            if e.response.status_code >= 400 and e.response.status_code != 404:
+                raise e
+            # Upload file if it doesn't exist.
+            with open(file_path, 'rb') as file:
+                content = file.read()
+                df = pd.DataFrame.from_records([{'content': content}])
```

Comment:
We should integrate chunking here. Can be alright for now but def one for after hackathon if we don't have time